src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -53,11 +54,22 @@ internal partial class Binder
             Debug.Assert(source != null);
             Debug.Assert((object)destination != null);
 
-            // We need to preserve any conversion that changes the type (even identity conversions, like object->dynamic),
-            // or that was explicitly written in code (so that GetSemanticInfo can find the syntax in the bound tree).
-            if (conversion.Kind == ConversionKind.Identity && !isCast && source.Type == destination)
+            if (conversion.IsIdentity)
             {
-                return source;
+                // identity tuple conversions result in a converted tuple
+                // to indicate that tuple conversions are no longer applicable.
+                if (source.Kind == BoundKind.TupleLiteral)
+                {
+                    var tuple = (BoundTupleLiteral)source;
+                    return new BoundConvertedTupleLiteral(tuple.Syntax, tuple.Arguments, tuple.ArgumentNamesOpt, tuple.Type, tuple.HasErrors);
```

Comment:
I think the result type of the node should be ```destination``` here. Also after we switch the node, we should probably continue because in case of ```isCast``` we likely want to create a node for conversion.